hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -364,9 +364,11 @@ public ResponseCodeEnum changeOwner(NftId nftId, AccountID from, AccountID to) {
 				return toFreezeAndKycValidity;
 			}
 
-			final var owner = (EntityId) nftsLedger.get(nftId, OWNER);
-			if (!owner.matches(from)) {
-				return SENDER_DOES_NOT_OWN_NFT_SERIAL_NO;
+			if (!nftId.isInternal()) {
```

Comment:
in the case for normal nft transfers, the `Nft` id will have `from`, `to` and `serialNumber` all valid. We should not include this check since `serialNumber` will never be `-1`.

We have to:
- get the owner of the nft
- if it is the default entity id -> verify that `from` is equal to the `token treasury`
- if it is not the default -> verify that `owner` matches `from`

Reply:
This code is also used for TokenUpdate operation where `serialNumber` is -1